bats/tests/preferences/verify-paths.bats

```diff
@@ -0,0 +1,99 @@
+# Test case 30
+
+load '../helpers/load'
+# Ensure subshells don't inherit a path that includes ~/.rd/bin
+export PATH=$(echo "$PATH" | tr ':' '\n' | grep -v /.rd/bin | tr '\n' ':')
+
+setup() {
+    if is_windows; then
+        skip "test not applicable on Windows"
+    fi
+}
+
+teardown_file() {
+    load '../helpers/load'
+    run rdctl shutdown
+    assert_nothing
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start app and fix dotfiles' {
+    start_container_engine
+    wait_for_container_engine
+    ensure_dotfiles_are_completed_BUG_BUG_BUG_4519
+}
+
+# Running `bash -l -c` can cause bats to hang, so close the output file descriptor with '3>&-'
+@test 'bash managed' {
+    if command -v bash >/dev/null; then
+        run bash -l -c "which rdctl" 3>&-
+        assert_success
+        assert_output --partial "$HOME/.rd/bin/rdctl"
+    else
+        skip 'bash not found'
+    fi
+}
+
+@test 'zsh managed' {
+    if command -v zsh >/dev/null; then
+        run zsh -i -c "which rdctl"
+        assert_success
+        assert_output --partial "$HOME/.rd/bin/rdctl"
+    else
+        skip 'zsh not found'
+    fi
+}
+
+@test 'fish managed' {
+    if command -v fish >/dev/null; then
+        run fish -c "which rdctl"
+        assert_success
+        assert_output --partial "$HOME/.rd/bin/rdctl"
+    else
+        skip 'fish not found'
+    fi
+}
+
+# This bashrc test assumes that this test will succeed, but it frees us
+# from sleeping after changing application.path-management-strategy
+no_bashrc_path_manager() {
+    run grep --silent 'MANAGED BY RANCHER DESKTOP START' "$HOME/.bashrc"
+    assert_failure
```

Comment:
I'm getting confused here by the use of `assert_failure` inside this function, but `assert_success` in the test after the `try` function calling this.

Given that both are checking the value of `$status` I'm not quite sure why the `assert_success` in the calling test isn't failing.

I would find it less confusing to write it like this:

```suggestion
    ! grep --silent 'MANAGED BY RANCHER DESKTOP START' "$HOME/.bashrc"
```

Reply:
Don't be confused -- I'm just not used to using `! COMMAND` in bash to test for failure